src/sentry/utils/committers.py

```diff
@@ -254,11 +246,24 @@ def get_serialized_event_file_committers(project, event, frame_limit=25):
 
     for committer in committers:
         commit_ids = [commit.id for (commit, _) in committer["commits"]]
-        committer["commits"] = [serialized_commits_by_id[commit_id] for commit_id in commit_ids]
+        commits_result = [serialized_commits_by_id[commit_id] for commit_id in commit_ids]
+        # Deduplicate commits
+        committer["commits"] = dedupeCommits(commits_result)
 
     metrics.incr(
         "feature.owners.has-committers",
         instance="hit" if committers else "miss",
         skip_internal=False,
     )
     return committers
+
+
+def dedupeCommits(commits):
+    seen = set()
+    result = []
```

Comment:
@NisanthanNanthakumar No need to use two different data structures here. You can just use a dictionary and then return `dict.values()`